extensions/resource-deployment/src/services/tools/toolBase.ts

```diff
@@ -19,14 +20,45 @@ export abstract class ToolBase implements ITool {
 	abstract type: ToolType;
 	abstract homePage: string;
 	abstract autoInstallSupported: boolean;
-	public abstract install(): Promise<void>;
+	abstract installationCommands: Command[];
 	protected abstract getVersionFromOutput(output: string): SemVer | undefined;
-	protected abstract readonly versionCommand: string;
+	protected abstract readonly versionCommand: Command;
+
+	protected get installationPath(): Promise<string | null> {
+		return new Promise<string | null>((resolve, _reject) => {
```

Comment:
It's preferred to use undefined over null (we mostly do undefined checking and I haven't seen a good reason to have null vs undefined)